addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -3246,7 +3246,7 @@ var SnippetsMenu = Widget.extend({
      */
     _onHideOverlay: function () {
         for (const editor of this.snippetEditors) {
-            editor.toggleOverlay(false);
+            editor.toggleOverlay(false, true);
```

Comment:
@utag-odoo I was asking for explanations and investigations, not another undocumented diff where I will ask questions too:
- If there is something that hides the overlay when using the crop feature, why do we need a new line to deactivate it?
- The crop widget code is independent from the "snippet" notion, it should not trigger such events on the bus
- As explained, in 14.0 there was no bug because image option were disabled after crop... I did not say it was the better behavior, just that it was not buggy. In any case: what made that behavior disappear between 14.0 and 15.0? Since you now restore it, you should have blamed/investigated around that?

Reply:
Hello @qsm-odoo,

> That previewMode param for the toggleOverlay function is kinda strange but I am pretty sure it should only be true when hovering the right panel or something like that.

the `previewMode` parameter defaults to true when hovering over the right panel.
Nevertheless, if we forcibly set it to true, it will consistently display the overlay,
contrary to our intended behavior.


> There are also multiple feature using that HideOverlay stuff, are they currently all broken in the same way?

Regarding the issue with `hide_overlay` functionality, it appears that it's currently experiencing a problem during transformations.
Additionally, we utilize `hide_overlay` in the context of `backgroundPosition,` but I'm not entirely certain if it's encountering the same issue there.
Our primary objective is to display the overlay exclusively after performing a crop operation.
Therefore, it's not to set `previewMode` to true, as this would aslo inadvertently show the `overlay` after `Transforming the picture.`

> If there is something that hides the overlay when using the crop feature, why do we need a new line to deactivate it?

It's my mistake here. To clarify, if there is code that hides the overlay when using the crop feature, we can achieve the desired outcome by simply removing that
```js 
this.trigger_up('hide_overlay');
```
instead of adding a new one to deactivate it.

> you should have blamed/investigated around that?

I found a commit [message](https://github.com/odoo/odoo/commit/5f41733eb7a40f6e335fdd257d984a3707e33469). In this commit, they removed the "deactivate_snippet" that was previously utilized when image options were disabled.

Reply:
Hello @qsm-odoo,
Can you please review this PR?